internal/restore/volumesnapshot_action.go

```diff
@@ -90,6 +90,22 @@ func (p *VolumeSnapshotRestoreItemAction) Execute(input *velero.RestoreItemActio
 			return nil, errors.Errorf("Volumesnapshot %s/%s does not have a %s annotation", vs.Namespace, vs.Name, util.CSIDriverNameAnnotation)
 		}
 
+		// if data mover is enabled, do not create VolumeSnapshotContent here
+		if util.DataMoverCase() {
+
+			vsMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&vs)
+			if err != nil {
+				return nil, errors.WithStack(err)
+			}
+
+			p.Log.Infof("Returning from VolumeSnapshotRestoreItemAction with no additionalItems")
+
+			return &velero.RestoreItemActionExecuteOutput{
```

Comment:
I maybe wrong here, should we not skip the restore of VS from here ? if we are restoring VS in the VSM plugin https://github.com/migtools/velero-plugin-for-vsm/pull/2/files

Reply:
If the restore is skipped here then the PVC restore fails, because that is not handled in the VSM plugin